### PR TITLE
add sql scripts for implement crud

### PR DESCRIPTION
**Pull request by gorr2012**

### CODE REVIEW

File: scripts/crud/course.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+-- CRUD for course
+
+-- CREATE
+INSERT INTO course (name, start_date, end_date) VALUES ('PHP', '2022-01-01T04:08:16.000Z', '2022-03-01T18:41:46.000Z');
+
+-- READ
+SELECT * FROM course WHERE id = 6
```
Comment (mikhama) [inline scripts/crud/course.sql]: ?

File: scripts/crud/course.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	c.id,
+	course_name,
+	c.start_date,
+	c.end_date
+
+-- UPDATE
+UPDATE course SET name='PHP and XAMP', start_date='2022-02-01T04:08:16.000Z', end_date='2022-04-01T18:41:46.000Z' WHERE id=6
+
+INSERT INTO course_student (course_id, student_id) VALUES (5, 101);
+INSERT INTO course_student (course_id, student_id) VALUES (4, 101);
```
Comment (mikhama) [inline scripts/crud/course.sql]: It's not related to updating the course, but student

File: scripts/crud/student.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
+-- CRUD for student
+
+-- CREATE
+INSERT INTO student (name, email, birth, origin) VALUES ('Rosemonde Pupkin', 'pupkin@pup.com', '1991-08-24T13:56:26.00Z', 'Izhevsk');
+
+-- READ
+SELECT * FROM student WHERE id = 101
```
Comment (mikhama) [inline scripts/crud/student.sql]: ?

File: scripts/crud/task.sql [hunk 4]
```diff
@@ -0,0 +1,32 @@
+-- CRUD for task
+
+-- CREATE
+INSERT INTO task (name, course_id, start_date, end_date)
+VALUES ('Variables JS', 1, '2022-01-10T04:08:16.000Z', '2022-03-11T18:41:46.000Z');
+
+-- READ
+SELECT * FROM task WHERE id = 51
```
Comment (mikhama) [inline scripts/crud/task.sql]: ?

File: scripts/crud/task.sql [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+SELECT
+	t.id,
+	t.name,
+	t.course_id,
```
Comment (mikhama) [inline scripts/crud/task.sql]: We don't need course_id here but course_name

File: scripts/crud/task.sql [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	c.name,
```
Comment (mikhama) [inline scripts/crud/task.sql]: You have t.name and c.name but where course and where task?

File: scripts/crud/task.sql [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	count(st.student_id) as students,
```
Comment (mikhama) [inline scripts/crud/task.sql]: distinct?

File: scripts/crud/task.sql [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+	array_agg(distinct s."name") as students_names
```
Comment (mikhama) [inline scripts/crud/task.sql]: Why do you wrap name in the quotes?

File: scripts/crud/task.sql (hunk 7 above)
Comment (mikhama) [inline scripts/crud/task.sql]: As far as I remember you need to count students that completed the task

File: scripts/crud/task.sql (hunk 8 above)
Comment (mikhama) [inline scripts/crud/task.sql]: Don't remember that I asked for student_names...

File: scripts/crud/student.sql [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	s.id,
+	s.name,
+	s.birth,
+	s.origin;
+
+-- UPDATE
+UPDATE student SET name= 'Rosemonde Zhopkin', email='zhopkin@pup.com', birth='1991-07-24T13:56:26.00Z', origin='Orsha' WHERE id=101
+
+INSERT INTO course_student (course_id, student_id) VALUES (5, 101);
+INSERT INTO course_student (course_id, student_id) VALUES (4, 101);
```
Comment (mikhama) [inline scripts/crud/student.sql:26]: You should probably fill in course_student_task here?

File: scripts/crud/task.sql [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+FROM task t
+JOIN course c ON c.id = t.course_id
+JOIN student_task st ON st.task_id = t.id
+JOIN student s ON s.id = st.student_id
```
Comment (mikhama) [inline scripts/crud/task.sql]: If there is a task that is not assigned to anyone - you'll not receive it using this query

File: scripts/crud/task.sql [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+	t.name,
+	t.course_id,
+	c.name,
+	count(st.student_id) as students,
+	array_agg(distinct s."name") as students_names
+FROM task t
+JOIN course c ON c.id = t.course_id
+JOIN student_task st ON st.task_id = t.id
+JOIN student s ON s.id = st.student_id 
+WHERE st.score > 0 AND st.finish_date <= t.end_date
```
Comment (mikhama) [inline scripts/crud/task.sql]: You won't receive the list of all the tasks this way

File: scripts/crud/student.sql [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+SELECT
+	s.id,
+	s.name,
+	s.birth,
+	s.origin,
+	array_agg(c.name) as courses
+FROM student s
+JOIN course_student cs ON s.id = cs.student_id 
+JOIN course c ON c.id = cs.course_id
```
Comment (mikhama) [inline scripts/crud/student.sql]: If student is not assigned to any course - you won't receive them

File: scripts/crud/course.sql [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+SELECT
+	c.id,
+	c.name as course_name,
+	c.start_date,
+	c.end_date,
+	count(distinct cs.student_id) as count_students,
+	array_agg(distinct t.name) as tasks
+FROM course c
+JOIN task t ON t.course_id = c.id
```
Comment (mikhama) [inline scripts/crud/course.sql]: If course is not contain any tasks you won't receive it

**Review by mikhama — CHANGES_REQUESTED** [review]

File: scripts/crud/student.sql (hunk 9 above)
Comment (mikhama) [reply scripts/crud/student.sql:26]:
```sql
INSERT INTO ...
SELECT
    t.course_id,
    t.id AS task_id
    101 as student_id
FROM task t
WHERE t.course_id = 5
```

Comment (mikhama) [reply scripts/crud/student.sql:26]: https://www.postgresql.org/docs/9.1/sql-insert.html

File: scripts/crud/task.sql (hunk 7 above)
Comment (mikhama) [inline scripts/crud/task.sql]: FILTER

File: scripts/crud/task.sql (hunk 11 above)
Comment (mikhama) [reply scripts/crud/task.sql]: score IS NOT NULL

Comment (mikhama) [reply scripts/crud/task.sql]: https://www.postgresql.org/docs/9.1/functions-comparison.html

Comment (mikhama) [reply scripts/crud/task.sql]: https://www.postgresql.org/docs/9.1/functions-comparisons.html

File: scripts/crud/task.sql (hunk 7 above)
Comment (mikhama) [reply scripts/crud/task.sql]: https://www.postgresql.org/docs/9.4/sql-expressions.html
4.2.7. Aggregate Expressions

File: scripts/crud/course.sql (hunk 1 above)
Comment (gorr2012) [reply scripts/crud/course.sql]: Deleted

File: scripts/crud/task.sql (hunk 4 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: Deleted

File: scripts/crud/course.sql (hunk 2 above)
Comment (gorr2012) [reply scripts/crud/course.sql]: deleted

File: scripts/crud/student.sql (hunk 3 above)
Comment (gorr2012) [reply scripts/crud/student.sql]: deleted

File: scripts/crud/task.sql (hunk 5 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: renamed

File: scripts/crud/task.sql (hunk 6 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: Separated

File: scripts/crud/task.sql (hunk 7 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: added

File: scripts/crud/task.sql (hunk 8 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: fixed

Comment (gorr2012) [reply scripts/crud/task.sql]: deleted

File: scripts/crud/course.sql (hunk 13 above)
Comment (gorr2012) [reply scripts/crud/course.sql]: Added left join

File: scripts/crud/student.sql (hunk 12 above)
Comment (gorr2012) [reply scripts/crud/student.sql]: added

File: scripts/crud/task.sql (hunk 10 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: added

File: scripts/crud/task.sql (hunk 7 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: done

File: scripts/crud/task.sql (hunk 11 above)
Comment (gorr2012) [reply scripts/crud/task.sql]: done

**Pull request merged**